zeebe/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ZeebePriorityDefinitionValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation.zeebe;
+
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebePriorityDefinition;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class ZeebePriorityDefinitionValidator
+    implements ModelElementValidator<ZeebePriorityDefinition> {
+
+  @Override
+  public Class<ZeebePriorityDefinition> getElementType() {
+    return ZeebePriorityDefinition.class;
+  }
+
+  @Override
+  public void validate(
+      final ZeebePriorityDefinition zeebePriorityDefinition,
+      final ValidationResultCollector validationResultCollector) {
+    final String priority = zeebePriorityDefinition.getPriority();
+
+    try {
+      final int priorityValue = Integer.parseInt(priority);
+      if (priorityValue < 0 || priorityValue > 100) {
```

Comment:
In regards to the decisions we took for this feature, for Camunda 7 parity, the range for the Priority of UserTasks is between 0 and 100.

The default value was decided to be 50 so as not to have any effect on the _end-user_ defined priorities in regards to sorting & filtering, as it falls in the middle of this range.

Reply:
I see! I also watched the recording from the kickoff discussion, and I understood from @tmetzke's point there that it's important for users to be able to create both higher and lower priority tasks since 50 is in the middle of the range